modules/library/jdbc/src/main/java/org/geotools/jdbc/JDBCDataStore.java

```diff
@@ -4147,17 +4242,35 @@ protected void encodeGroupByStatement(
 
     protected void encodeFunction(
             String function, Expression expression, StringBuffer sql, FilterToSQL filterToSQL) {
-        sql.append(function).append("(");
-        if (expression == null) {
+        encodeFunction(function, Arrays.asList(expression), sql, filterToSQL);
+    }
+
+    protected void encodeFunction(
+            String function,
+            List<Expression> expressions,
+            StringBuffer sql,
+            FilterToSQL filterToSQL) {
+        sql.append(function);
```

Comment:
It seems that for no expression or single expression this would work:
* function(*)
* function(myAtt)
but for multiple attributes it would end up encoding something like:
* function(att1),(att2),(att3)

Uh?? :-D

Reply:
ops... fixed